tests/artifacts_tests/test_download_artifact.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import pathlib
+
+import pytest
+
+import optuna
+from optuna.artifacts import download_artifact
+from optuna.artifacts import FileSystemArtifactStore
+from optuna.artifacts import upload_artifact
+from optuna.artifacts._protocol import ArtifactStore
+
+
+@pytest.fixture(params=["FileSystem"])
+def artifact_store(tmp_path: pathlib.PurePath, request: pytest.FixtureRequest) -> ArtifactStore:
+    if request.param == "FileSystem":
+        return FileSystemArtifactStore(str(tmp_path))
+    assert False, f"Unknown artifact store: {request.param}"
+
+
+def test_download_artifact(tmp_path: pathlib.PurePath, artifact_store: ArtifactStore) -> None:
+    storage = optuna.storages.InMemoryStorage()
+    study = optuna.create_study(storage=storage)
```

Comment:
`InMemoryStorage` is the default storage, and it can be omitted.

```suggestion
    study = optuna.create_study()
```